### PR TITLE
Add `AsyncSeekForwardExt` trait to allows a similar API to the previous Bevy version

### PR DESCRIPTION
**Pull request by ldubos**
# Objective

This PR introduces an `AsyncSeekForwardExt` trait, which I forgot in my previous PR #14194.

This new trait is analogous to `AsyncSeekExt` and allows all implementors of `AsyncSeekForward` to directly use the `seek_forward` function in async contexts.

## Solution

- Implement a new `AsyncSeekForwardExt` trait
- Automatically implement this trait for all types that implement `AsyncSeekForward`

## Showcase

This new trait allows a similar API to the previous Bevy version:

```rust
#[derive(Default)]
struct UniverseLoader;

#[derive(Asset, TypePath, Debug)]
struct JustALilAsteroid([u8; 128]);

impl AssetLoader for UniverseLoader {
    type Asset = JustALilAsteroid;
    type Settings = ();
    type Error = std::io::Error;

    async fn load<'a>(
        &'a self,
        reader: &'a mut Reader<'a>,
        _settings: &'a Self::Settings,
        _context: &'a mut LoadContext<'_>,
    ) -> Result<Self::Asset, Self::Error> {
        // read the asteroids entry table
        let entry_offset: u64 = /* ... */;
        let current_offset: u64 = reader.seek_forward(0).await?;

        // jump to the entry
        reader.seek_forward(entry_offset - current_offset).await?;

        let mut asteroid_buf = [0; 128];
        reader.read_exact(&mut asteroid_buf).await?;

        Ok(JustALilAsteroid(asteroid_buf))
    }
    
    fn extensions(&self) -> &[&str] {
        &["celestial"]
    }
}
```

### CODE REVIEW

**Review by BenjaminBrienen — APPROVED** [review]

File: crates/bevy_asset/src/io/mod.rs [hunk 1]
```diff
@@ -120,6 +121,38 @@ impl<T: ?Sized + AsyncSeekForward + Unpin> AsyncSeekForward for Box<T> {
     }
 }
 
+pub trait AsyncSeekForwardExt: AsyncSeekForward {
+    fn seek_forward(&mut self, offset: u64) -> SeekForwardFuture<'_, Self>
```
Comment (bushrat011899) [inline crates/bevy_asset/src/io/mod.rs:125]: Should include some documentation here. E.g.,

```rust
/// Seek by the provided `offset` in the forwards direction, using the [`AsyncSeekForward`] trait.
```

**Review by bushrat011899 — APPROVED** [review]
Looks good with a clean implementation. I do recommend putting some documentation on the method though, even if the method name is self explanatory.

File: crates/bevy_asset/src/io/mod.rs (hunk 1 above)
Comment (ldubos) [reply crates/bevy_asset/src/io/mod.rs:125]: Good point

**Review by cart — APPROVED** [review]
Definitely good to make this easy to call!

**Pull request merged**